src/threadinterrupt.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <mutex>
+#include <condition_variable>
+#include <atomic>
+
+class CThreadInterrupt
```

Comment:
Sometimes you'll want a real interruptible condvar, though. See https://github.com/theuni/bitcoin/commit/da68e3cf5832878fbb1951eb1eeb01d8cd22d1e1#diff-9a82240fe7dfe86564178691cc57f2f1R1887.

The alternative would be to add wait_for/wait_until/notify_one/notify_all to CThreadInterrupt, then just forward them to the condvar. But for that, you'd still need to expose the mutex.